File: androidtoolkit/src/main/java/net/oschina/git/roland/androidtoolkit/validation/BaseValidationTask.java
package net.oschina.git.roland.androidtoolkit.validation;

import android.util.Log;

/**
 * 基础的校验任务类
 */
public abstract class BaseValidationTask implements Runnable {

    private static final String TAG = BaseValidationTask.class.getSimpleName();

    protected String name;

    protected boolean isRunning = false;

    protected boolean finished = false;

    protected boolean result = false;

    protected int failCode = -1;

    protected OnValidationFinishListener onValidationFinishListener;

    public BaseValidationTask(String name) {
        this.name = name;
    }

    protected void finish(boolean valid) {
        Log.d(TAG, "task " + name + "finish result = " + String.valueOf(valid));
        result = valid;
        finished = true;
        isRunning = false;
        if (onValidationFinishListener != null) {
            onValidationFinishListener.onTaskFinish(this);
        }
    }

    public void setOnValidationFinishListener(OnValidationFinishListener listener) {
        onValidationFinishListener = listener;
    }


    public String getName() {
        return name;
    }

    public boolean getResult() {
        return result;
    }

    public boolean isRunning() {
        return !isRunning;
    }

    public boolean finished() {
        return finished;
    }

    public int getFailCode() {
        return failCode;
    }
}

File: androidtoolkit/src/main/java/net/oschina/git/roland/androidtoolkit/validation/BaseValidationTask.kt
package net.oschina.git.roland.androidtoolkit.validation

import android.util.Log

/**
 * 基础的校验任务类
 */
abstract class BaseValidationTask(var name: String) : Runnable {

    protected var bRunning = false

    protected var finished = false

    var result = false
        protected set

    var failCode = -1
        protected set

    protected var validationFinishListener: OnValidationFinishListener? = null

    protected fun finish(valid: Boolean) {
        Log.d(TAG, "task " + name + " finish result = " + valid.toString())
        result = valid
        finished = true
        bRunning = false
        if (validationFinishListener != null) {
            validationFinishListener!!.onTaskFinish(this)
        }
    }

    fun isRunning(): Boolean {
        return !bRunning
    }

    fun finished(): Boolean {
        return finished
    }

    fun setOnValidationFinishListener(listener: OnValidationFinishListener) {
        validationFinishListener = listener
    }

    companion object {

        private val TAG = BaseValidationTask::class.java.simpleName
    }
}
